plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/product-collection-content.tsx

```diff
@@ -17,17 +22,68 @@ import type {
 	ProductCollectionAttributes,
 	ProductCollectionQuery,
 	ProductCollectionEditComponentProps,
+	PreviewState,
+	SetPreviewState,
 } from '../types';
 import { DEFAULT_ATTRIBUTES, INNER_BLOCKS_TEMPLATE } from '../constants';
 import { getDefaultValueOfInheritQueryFromTemplate } from '../utils';
 import InspectorControls from './inspector-controls';
 import InspectorAdvancedControls from './inspector-advanced-controls';
 import ToolbarControls from './toolbar-controls';
 
-const ProductCollectionContent = (
-	props: ProductCollectionEditComponentProps
-) => {
+const useSetPreviewState = ( {
+	setAttributes,
+	location,
+	setPreviewState,
+	attributes,
+}: {
+	setAttributes: (
+		attributes: Partial< ProductCollectionAttributes >
+	) => void;
+	location: WooCommerceBlockLocation;
+	attributes: ProductCollectionAttributes;
+	setPreviewState?: SetPreviewState | undefined;
+} ) => {
+	const setState = ( newPreviewState: PreviewState ) => {
+		setAttributes( {
+			...attributes.previewState,
+			previewState: newPreviewState,
+		} );
```

Comment:
```suggestion
		setAttributes( {
			...attributes,
			previewState: newPreviewState,
		} );
```

I know it's just POC but pointing out you probably meant this ☝️ so you don't copy-paste it to final solution. Unless I don't understand something and `previewState` is restructured for purpose.

Reply:
Ah! Good catch. I actually meant this (see below) in case users don't provide all attributes while using `setState`. 
```tsx
setAttributes( {
			previewState: {
				...attributes.previewState,
				...newPreviewState,
			},
		} );
```

Reply:
Made the changes in https://github.com/woocommerce/woocommerce/pull/45703/commits/f17a6949b1dd8e3500d27ea559416476ab9f21ba